packages/smooth_app/lib/pages/preferences_v2/preferences_page.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_page.dart';
+import 'package:smooth_app/pages/preferences_v2/app_bars/logged_in_app_bar.dart';
+import 'package:smooth_app/pages/preferences_v2/app_bars/logged_out_app_bar.dart';
+import 'package:smooth_app/pages/preferences_v2/cards/preference_card.dart';
+import 'package:smooth_app/pages/preferences_v2/roots/app_settings_root.dart';
+import 'package:smooth_app/pages/preferences_v2/roots/default_root.dart';
+import 'package:smooth_app/pages/preferences_v2/roots/preferences_root.dart';
+import 'package:smooth_app/pages/preferences_v2/tiles/navigation_preference_tile.dart';
+import 'package:smooth_app/pages/preferences_v2/tiles/preference_tile.dart';
+import 'package:smooth_app/pages/preferences_v2/tiles/url_preference_tile.dart';
+
+class PreferencesPage extends StatelessWidget {
+  String? _getUserId() => OpenFoodAPIConfiguration.globalUser?.userId;
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+
+    return ChangeNotifierProvider<PreferencesRootSearchController>(
+      create: (_) => PreferencesRootSearchController(),
+      child: DefaultPreferencesRoot(
+        customAppBar: _getUserId() != null
+            ? const LoggedInAppBar()
+            : const LoggedOutAppBar(),
```

Comment:
```suggestion
```suggestion
  @override
  Widget build(BuildContext context) {
    final AppLocalizations appLocalizations = AppLocalizations.of(context);
    final String? userId = OpenFoodAPIConfiguration.globalUser?.userId;


    return ChangeNotifierProvider<PreferencesRootSearchController>(
      create: (_) => PreferencesRootSearchController(),
      child: DefaultPreferencesRoot(
        customAppBar: userId != null
            ? const LoggedInAppBar(userId)
            : const LoggedOutAppBar(),
```